### PR TITLE
Add `UnobservedComponents` statsmodels wrapper

### PR DESCRIPTION
**Pull request by juanitorduz**
#### Reference Issues/PRs

Fixes https://github.com/alan-turing-institute/sktime/issues/1371

#### What does this implement/fix? Explain your changes.

This PR wraps the [UnobservedComponents](https://www.statsmodels.org/stable/generated/statsmodels.tsa.statespace.structural.UnobservedComponents.html) from statsmodels. We follow closely the existing implementation [sktime/forecasting/ets.py](https://github.com/alan-turing-institute/sktime/blob/main/sktime/forecasting/ets.py) using the [` _StatsModelsAdapter`](https://github.com/alan-turing-institute/sktime/blob/main/sktime/forecasting/base/adapters/_statsmodels.py) class (a minor change was added in order to support `exog` variables).

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### What should a reviewer concentrate their feedback on?

In this initial PR I provide a basic implementation of the wrapper. Still, I wanna make sure I understand what are the requirements the code needs in order to merge to `main`. Concretely here are my questions:

**Main Questions:**

- The docstrings of the original statsmodels class [UnobservedComponents](https://www.statsmodels.org/stable/generated/statsmodels.tsa.statespace.structural.UnobservedComponents.html) is very complete and gives a clear description of the model and the `level` variable which determines the structure of the state space model. How much should I add to the doctrings on the `sktime` implementation?

- Which kind of  test should I write? Probably a couple of test ensuring the statsmodels implementation agrees with this one.

- Should I add this model to the [`examples/01_forecasting.ipynb`](https://github.com/alan-turing-institute/sktime/blob/main/examples/01_forecasting.ipynb) notebook? Any other place where I should add more examples on this new estimator?

- Any other feedback or suggestion?

**Minor Questions**

- I wanted to add some useful plotting functions to this wrapper. Unfortunately the simple method `plot_components` fils because a `TypeError: float() argument must be a string or a number, not 'Period'`. Have you encounter this type of error in the past? It seems the internal plotting functions do not handle `Period` types.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [x] I've added unit tests and made sure they pass locally.

##### For new estimators
- [x] I've added complete doc-strings.
- [x] I've added the estimator to the online documentation.
- [x] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by juanitorduz** [comment]
In https://github.com/alan-turing-institute/sktime/pull/1394/commits/c6180a7cdd38f63c4238a676df2f46dbe33183ea I added the first test to compare with the original implementation from `statsmodels` (plus an additional docs fix https://github.com/alan-turing-institute/sktime/pull/1394/commits/4129b77d9bf79b715a62ef2482554256f69b0b26)

**Comment by juanitorduz** [comment]
In https://github.com/alan-turing-institute/sktime/pull/1394/commits/68f920d223be75546836085ec9da2f48827aba45 I added the `UnobservedComponets` estimator into the `01_forecasting` notebook (section 2.5)

**Comment by juanitorduz** [comment]
In https://github.com/alan-turing-institute/sktime/pull/1394/commits/884828af27c7cdcd80840b9ac2d30d20dee7fea9 I added the model into the online documentation. 

I think this draft PR is ready for a first review from the maintainers :)

**Comment by juanitorduz** [comment]
In https://github.com/alan-turing-institute/sktime/pull/1394/commits/95ccf6e82e48a0f68690b1554c34c64a13d5f041 I added the doc-strings. Mostly taken from the ones in statsmodels. Again, I am not sure how much information should be added. Feedback is welcome :)

File: sktime/forecasting/structural.py
Comment (mloning) [inline sktime/forecasting/structural.py:7]: We try to use GitHub handles, that way we can eventually link them in our [estimator overview table](https://www.sktime.org/en/latest/estimator_overview.html)
```suggestion
__author__ = ["juanitorduz"]
```

File: sktime/forecasting/tests/test_structural.py
Comment (mloning) [inline sktime/forecasting/tests/test_structural.py:3]:
```suggestion
__author__ = ["juanitorduz"]
```

**Review by mloning — COMMENTED** [review]
Hi @juanitorduz, I had a quick look and I think this looks very good! Only two minor comments. I haven't looked at the notebook changes yet.

File: docs/source/api_reference/forecasting.rst
Comment (mloning) [inline docs/source/api_reference/forecasting.rst:130]: since this is just the section header I'd write it like
```suggestion
Unobserved Components
```

File: sktime/forecasting/structural.py
Comment (mloning) [inline sktime/forecasting/structural.py:172]: These links aren't rendered as hyperlinks (see https://sktime--1394.org.readthedocs.build/en/1394/api_reference/auto_generated/sktime.forecasting.structural.UnobservedComponents.html#sktime.forecasting.structural.UnobservedComponents, compare e.g. the See Also section of the [CosineTransformer](https://www.sktime.org/en/latest/api_reference/auto_generated/sktime.transformations.series.cos.CosineTransformer.html)). 

To fix this, I think we need to add statsmodels to the intersphinx specification here: https://github.com/alan-turing-institute/sktime/blob/3b567d315766878009e4f3b24bfc19d8af020144/docs/source/conf.py#L421

Comment (juanitorduz) [reply sktime/forecasting/structural.py]: I tried in https://github.com/alan-turing-institute/sktime/pull/1394/commits/c81e9143ad265ad454fa86dafcea8da0f54357b3 to follow your suggestion.

Comment (juanitorduz) [reply sktime/forecasting/structural.py]: It looks as expected [here](https://sktime--1394.org.readthedocs.build/en/1394/api_reference/auto_generated/sktime.forecasting.structural.UnobservedComponents.html#sktime.forecasting.structural.UnobservedComponents) right?

Comment (juanitorduz) [reply sktime/forecasting/structural.py]: Mmm I do not understand why some test-examples are failing in https://github.com/alan-turing-institute/sktime/pull/1394/checks?check_run_id=3580158874  as it seems relates to some `k-means` which is not in the scope of the changes of this PR ... ❓

File: sktime/forecasting/structural.py
Comment (mloning) [inline sktime/forecasting/structural.py:173]: I think without the `func` it should turn them into hyperlinks, possibly
```suggestion
    statsmodels.tsa.statespace.structural.UnobservedComponents
    statsmodels.tsa.statespace.structural.UnobservedComponentsResults
```

Comment (juanitorduz) [reply sktime/forecasting/structural.py]: Thanks! In addition I corrected the link in https://github.com/alan-turing-institute/sktime/pull/1394/commits/c2bffc00eb6a9c2112e1d9aea8196c1c41ce7c52 and now it works locally as expected 🚀 !

**Comment by juanitorduz** [comment]
@mloning I do not understand why the Azure pipelines are failing 😢 ... any tip?

**Review by mloning — APPROVED** [review]
All looks good to me now! The links now also reader correctly and directly link to the statsmodels online docs! Thanks @juanitorduz 

The Azure pipeline failures were due to #1415 but all resolved now.

I'll merge in the next couple of days if nobody else comments.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Great addition!

However, there are - probably accidentally - massive changes to the forecasting notebook!
These should be reverted before we merge.

**Comment by juanitorduz** [comment]
> Great addition!
> 
> However, there are - probably accidentally - massive changes to the forecasting notebook!
> These should be reverted before we merge.

Ups! by looking into https://app.reviewnb.com/alan-turing-institute/sktime/pull/1394/files/ I see that in the `master` branch the notebook does not have any cell outputs 🙈 . I will fix this.

**Comment by juanitorduz** [comment]
> > Great addition!
> > However, there are - probably accidentally - massive changes to the forecasting notebook!
> > These should be reverted before we merge.
> 
> Ups! by looking into https://app.reviewnb.com/alan-turing-institute/sktime/pull/1394/files/ I see that in the `master` branch the notebook does not have any cell outputs 🙈 . I will fix this.

By looking into the diff and https://app.reviewnb.com/alan-turing-institute/sktime/pull/1394/files/ it seems https://github.com/juanitorduz/sktime/commit/f2f939cf7461dd0cb4e4450deef00555aae69c1f and https://github.com/juanitorduz/sktime/commit/9ff3d3f3abe88b45a3e9e08ffa80506507a7f358 solved the problem. 

**Remark:** I had to clear the notebook output twice: (1) The first one in VSCode (my IDE) which actually changed a bit the notebook structure (json order?) and then (2) restart using Jupyter notebooks (Anaconda), which solved the "high" number of changes issue 😃 !

**Review by fkiraly — APPROVED** [review]
Yes, thanks for fixing the notebook.

Unrelated, we may later want to add the automated estimator list to section 2, using `all_estimators` - though that's better a separate PR.

**Pull request merged**
